src/fides/api/ops/task/graph_task.py

```diff
@@ -553,6 +554,30 @@ def erasure_request(self, retrieved_data: List[Row], *inputs: List[Row]) -> int:
         )  # Cache that the erasure was performed in case we need to restart
         return output
 
+    @retry(action_type=ActionType.consent, default_return=False)
+    def consent_request(self, identity: Dict[str, Any]) -> bool:
+        """Run consent request request"""
+
+        if not self.can_write_data():
+            logger.warning(
+                "No consent on {} as its ConnectionConfig does not have write access.",
+                self.traversal_node.node.address,
+            )
+            self.update_status(
+                f"No values were erased since this connection {self.connector.configuration.key} has not been "
+                f"given write access",
+                None,
+                ActionType.erasure,
+                ExecutionLogStatus.error,
+            )
+            return False
+
+        output: bool = self.connector.run_consent_request(
+            self.traversal_node, self.resources.policy, self.resources.request, identity
+        )
+        self.log_end(ActionType.consent)
```

Comment:
The parallel GraphTask.erasure_request (and access_request) also caches that the erasure was completed on this node so if we restart from failure, we don't re-attempt this node.   This is used to update the graph on retry.

```
self.resources.cache_erasure(
            f"{self.key}", output
        ) 
```

There's a lot of good reasons to not try to re-run an erasure request on a collection if it's already been completed, but I don't know if we need that level of detail here, beyond the execution logs.  I think if a consent request fails, we could run the entire step in its entirety. `cache_failed_checkpoint_details` caches the step where it failed, but I don't think we need to build a modified graph on retry.